File: api-code-examples/swift/doc-get.swift
import IrohLib

let node = try! IroneNode()
let author = try! node.createAuthor()
let doc = try! Iroh.createDoc()

let hash = try! doc.setBytes(author: author, key: "foo".data(using: .utf8)!, value: "bar".data(using: .utf8)!)
let value = try! doc.getContentBytes(hash: hash)

File: api-code-examples/go/doc-get.go
package main

import (
	"fmt"

	"github.com/n0-computer/iroh-ffi/iroh-go/iroh"
)

func main() {
	node, err := iroh.NewIrohNode("iroh_data_dir")
	if err != nil {
		// real programs handle errors!
		panic(err)
	}

	author, err := node.AuthorNew()
	if err != nil {
		panic(err)
	}
	fmt.Printf("Created author %s\n", author.ToString())

	doc, err := node.DocNew()
	if err != nil {
		panic(err)
	}
	fmt.Printf("Created document %s\n", doc.Id())

	hash, err := doc.SetBytes(author, []byte("go"), []byte("says hello"))
	if err != nil {
		panic(err)
	}
	fmt.Printf("Inserted %s\n", hash.ToString())

	content, err := doc.GetContentBytes(hash)
	if err != nil {
		panic(err)
	}
	fmt.Printf("Got content %q\n", string(content))
}
